pages/docs/core-concepts/create-and-run-effects.mdx

```diff
@@ -106,48 +107,99 @@ Once again inspecting the return type we can see that it is `Effect<never, never
+  both faster than `Promise`-based APIs as well as more precise in handling
+  failures.
 </Callout>
 
 As an example, we can demonstrate wrapping `readFile` from the NodeJS `fs` module with Effect:
 
 ```ts twoslash
-import * as Effect from "@effect/io/Effect"
-import * as NodeFS from "node:fs"
+import * as Effect from "@effect/io/Effect";
+import * as NodeFS from "node:fs";
 
-export const readTodos = Effect.async<never, NodeJS.ErrnoException, Buffer>((resume) => {
-//           ^?
-  NodeFS.readFile("todos.txt", (error, data) => {
-    if (error) {
-      resume(Effect.fail(error))
-    } else {
-      resume(Effect.succeed(data))
-    }
-  })
-})
+export const readTodos = Effect.async<never, NodeJS.ErrnoException, Buffer>(
+  //              ^?
+  (resume) => {
+    NodeFS.readFile("todos.txt", (error, data) => {
+      if (error) {
+        resume(Effect.fail(error));
+      } else {
+        resume(Effect.succeed(data));
+      }
+    });
+  }
+);
 ```
 
-Something to note in the above example is that we were actually forced to manually annotate the types when calling `Effect.async`. Unfortunately, TypeScript is not capable of inferring the type parameters for a callback based on what is returned inside the body of the callback. This limitation of TypeScript is the one minor drawback of using callback-based APIs.
+Something to note in the above example is that we were actually forced to manually annotate the types when calling `Effect.async`. Unfortunately, TypeScript is not capable of inferring the type parameters for a callback based on what is returned inside the body of the callback. This limitation of TypeScript is the one minor drawback of using callback-based APIs. However, it will validate that the types you passed in match what is being returned.
 
 Another thing to note is that we probably don't want to have to call the `Effect.async` constructor everywhere that we use callback-based APIs. Therefore, it is a very common pattern to define helpers which wrap external APIs into an Effect-based API.
 
 For example, we can refactor the previous code into:
 
 ```ts twoslash
-import * as Effect from "@effect/io/Effect"
-import * as NodeFS from "node:fs"
+import * as Effect from "@effect/io/Effect";
+import * as NodeFS from "node:fs";
 
-export const readFile = (path: string): Effect.Effect<never, NodeJS.ErrnoException, Buffer> =>
+export const readFile = (
+  path: string
+): Effect.Effect<never, NodeJS.ErrnoException, Buffer> =>
   Effect.async((resume) => {
     NodeFS.readFile(path, (error, data) => {
       if (error) {
-        resume(Effect.fail(error))
+        resume(Effect.fail(error));
       } else {
-        resume(Effect.succeed(data))
+        resume(Effect.succeed(data));
       }
-    })
-  })
+    });
+  });
+
+export const readTodos = readFile("todos.txt");
+//           ^?
+```
 
-export const readTodos = readFile("todos.txt")
+## Running Effects
```

Comment:
In all honesty, we could consider moving this to a dedicated section since there are various ways of running Effects and they all have slightly different implications. 

I think it may be better to go ahead and do that - let's create a section called "Running Effects" in the sidebar.

Reply:
I was considering this as well, I initially didn't have enough content so I put them in the same place, but I think we can separate them and add content later.

Reply:
Anything else you think we should add in here at this second? Or should we merge and revisit?